### PR TITLE
[receiver/tcp|udplog] Migrate config unmarshaling from yaml to mapstructure

### PR DESCRIPTION
**Pull request by djaglowski**
pkg/stanza historically used yaml for configuration unmarshaling.
The process of migrating to mapstructure was begun long ago but
was never completed. This PR switches tcplog & udplog underlying operators
to use mapstructure unmarshaling.

### CODE REVIEW

File: receiver/udplogreceiver/udp_test.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/input/udp"
```
Comment (dmitryax) [inline receiver/udplogreceiver/udp_test.go:25]: Hm.. this must be part of the third import group. I'm curios why linter wan't triggered

**Review by dmitryax — COMMENTED** [review]
Just one nit

**Comment by dmitryax** [comment]
Please resolve the conflict

File: receiver/udplogreceiver/udp_test.go (hunk 1 above)
Comment (djaglowski) [reply receiver/udplogreceiver/udp_test.go:25]: Yeah, strange. Looking into it.

Comment (djaglowski) [reply receiver/udplogreceiver/udp_test.go]: I opened #13318 to track

**Review by dmitryax — APPROVED** [review]

**Pull request merged**